x/stakeibc/keeper/icacallbacks_detokenize.go

```diff
@@ -46,12 +46,23 @@ func DetokenizeCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, a
 	k.Logger(ctx).Info(utils.LogICACallbackStatusWithHostZone(chainId, ICACallbackID_Detokenize,
 		icacallbackstypes.AckResponseStatus_SUCCESS, packet))
 
+	// Confirm host zone exists
+	hostZone, found := k.GetHostZone(ctx, chainId)
+	if !found {
+		return errorsmod.Wrapf(types.ErrHostZoneNotFound, "Host zone not found: %s", chainId)
+	}
+
 	// If the ICA succeeded, remove the token deposit
 	deposit := detokenizeCallback.Deposit
 	k.RecordsKeeper.RemoveLSMTokenDeposit(ctx, deposit.ChainId, deposit.Denom)
 
 	// Update delegation on the host zone and validator
-	// TODO [LSM] - Use helper function
+	err := k.AddDelegationToValidator(ctx, hostZone, deposit.ValidatorAddress, deposit.Amount, ICACallbackID_Detokenize)
+	if err != nil {
+		return err
+	}
+	hostZone.TotalDelegations = hostZone.TotalDelegations.Add(deposit.Amount)
```

Comment:
Wdyt about putting this update inside `AddDelegationToValidator`? I think we would always want to update them in tandem, but might be overlooking an edge case

Reply:
I don't think we need it now. Just putting it on the radar!